src/vs/workbench/contrib/positronPlots/browser/positronPlotsService.ts

```diff
@@ -164,6 +183,34 @@ export class PositronPlotsService extends Disposable implements IPositronPlotsSe
 			}
 		});
 
+		// When the extension service is about to stop, remove any HTML plots
+		// from the plots list. These plots are backed by a proxy that runs in
+		// the extension host, so may become invalid when the extension host is
+		// stopped.
+		this._register(this._extensionService.onWillStop((e) => {
+			// Nothing to do if there are no plots
+			if (this._plots.length === 0) {
+				return;
+			}
+			let removedSelectedPlot = false;
+			this._plots.forEach((plot) => {
+				if (plot instanceof HtmlPlotClient) {
+					plot.dispose();
+					if (this._selectedPlotId === plot.id) {
+						removedSelectedPlot = true;
+					}
+					this._plots.splice(this._plots.indexOf(plot), 1);
+				}
+			});
+
+			this._onDidReplacePlots.fire(this._plots);
+
+			// If we removed the selected plot, select the first plot in the list
+			if (removedSelectedPlot && this._plots.length > 0) {
+				this.selectPlot(this._plots[0].id);
+			}
```

Comment:
I wonder whether we should select the nearest plot to the selected plot that was removed?

Reply:
A good question ... I think that'd be better but would make this a lot more complicated. Practically speaking I don't think this will happen much as the extension host doesn't get stopped (other than for a full shutdown) during the normal course of operation; this is really to handle extension host crashes, etc.